test/Sentry.Extensions.Logging.Tests/SentrySqlListenerTests.cs

```diff
@@ -0,0 +1,242 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using NSubstitute;
+using Xunit;
+
+namespace Sentry.Extensions.Logging.Tests
+{
+    public class SentrySqlListenerTests
+    {
+        internal const string SqlDataWriteConnectionOpenBeforeCommand = SentrySqlListener.SqlDataWriteConnectionOpenBeforeCommand;
+        internal const string SqlMicrosoftWriteConnectionOpenBeforeCommand = SentrySqlListener.SqlMicrosoftWriteConnectionOpenBeforeCommand;
+
+        internal const string SqlMicrosoftWriteConnectionOpenAfterCommand = SentrySqlListener.SqlMicrosoftWriteConnectionOpenAfterCommand;
+        internal const string SqlDataWriteConnectionOpenAfterCommand = SentrySqlListener.SqlDataWriteConnectionOpenAfterCommand;
+
+        internal const string SqlMicrosoftWriteConnectionCloseAfterCommand = SentrySqlListener.SqlMicrosoftWriteConnectionCloseAfterCommand;
+        internal const string SqlDataWriteConnectionCloseAfterCommand = SentrySqlListener.SqlDataWriteConnectionCloseAfterCommand;
+
+        internal const string SqlDataBeforeExecuteCommand = SentrySqlListener.SqlDataBeforeExecuteCommand;
+        internal const string SqlMicrosoftBeforeExecuteCommand = SentrySqlListener.SqlMicrosoftBeforeExecuteCommand;
+
+        internal const string SqlDataAfterExecuteCommand = SentrySqlListener.SqlDataAfterExecuteCommand;
+        internal const string SqlMicrosoftAfterExecuteCommand = SentrySqlListener.SqlMicrosoftAfterExecuteCommand;
+
+        internal const string SqlDataWriteCommandError = SentrySqlListener.SqlDataWriteCommandError;
+        internal const string SqlMicrosoftWriteCommandError = SentrySqlListener.SqlMicrosoftWriteCommandError;
+
+        private Func<ISpan, bool> GetValidator(string type) =>
+    type switch
+    {
+        var x when
+                x == SqlDataWriteConnectionOpenBeforeCommand ||
+                x == SqlMicrosoftWriteConnectionOpenBeforeCommand ||
+                x == SqlMicrosoftWriteConnectionOpenAfterCommand ||
+                x == SqlDataWriteConnectionOpenAfterCommand ||
+                x == SqlMicrosoftWriteConnectionCloseAfterCommand ||
+                x == SqlDataWriteConnectionCloseAfterCommand
+            => (span) => span.Description is null && span.Operation == "db.connection",
+        var x when
+                x == SqlDataBeforeExecuteCommand ||
+                x == SqlMicrosoftBeforeExecuteCommand ||
+                x == SqlDataAfterExecuteCommand ||
+                x == SqlMicrosoftAfterExecuteCommand ||
+                x == SqlDataWriteCommandError ||
+                x == SqlMicrosoftWriteCommandError
+            => (span) => span.Operation == "db.query",
+        _ => throw new NotSupportedException()
+    };
```

Comment:
```suggestion
		        type switch
		        {
		            var x when
		                    x == SqlDataWriteConnectionOpenBeforeCommand ||
		                    x == SqlMicrosoftWriteConnectionOpenBeforeCommand ||
		                    x == SqlMicrosoftWriteConnectionOpenAfterCommand ||
		                    x == SqlDataWriteConnectionOpenAfterCommand ||
		                    x == SqlMicrosoftWriteConnectionCloseAfterCommand ||
		                    x == SqlDataWriteConnectionCloseAfterCommand
		                => (span) => span.Description is null && span.Operation == "db.connection",
		            var x when
		                    x == SqlDataBeforeExecuteCommand ||
		                    x == SqlMicrosoftBeforeExecuteCommand ||
		                    x == SqlDataAfterExecuteCommand ||
		                    x == SqlMicrosoftAfterExecuteCommand ||
		                    x == SqlDataWriteCommandError ||
		                    x == SqlMicrosoftWriteCommandError
		                => (span) => span.Operation == "db.query",
		            _ => throw new NotSupportedException()
		        };
```

Reply:
Also, please don't name things `x`. Name it the first character of the thing you're dealing with, or the whole name.